### PR TITLE
Allow spikegadgets to handle another data style

### LINKED ISSUE #1517: read_spikegadgets broken for neuropixel recordings
Hi,
The  read_spikegadgets from spikeinterface was working with neuropixel probes with this committ 83a84b2 but now it is broken.
```
 File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\IPython\core\interactiveshell.py", line 3550, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-3052bc77e239>", line 1, in <module>
    raw_rec = read_spikegadgets(r"X:\13\ephys\20231213_155419.rec\20231213_155419.rec" )
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\spikeinterface\extractors\neoextractors\spikegadgets.py", line 45, in __init__
    NeoBaseRecordingExtractor.__init__(
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\spikeinterface\extractors\neoextractors\neobaseextractor.py", line 188, in __init__
    _NeoBaseExtractor.__init__(self, block_index, **neo_kwargs)
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\spikeinterface\extractors\neoextractors\neobaseextractor.py", line 27, in __init__
    self.neo_reader = self.get_neo_io_reader(self.NeoRawIOClass, **neo_kwargs)
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\spikeinterface\extractors\neoextractors\neobaseextractor.py", line 66, in get_neo_io_reader
    neo_reader.parse_header()
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\neo\rawio\baserawio.py", line 189, in parse_header
    self._parse_header()
  File "C:\Users\SciencePerson\miniforge3\envs\spikesorting\lib\site-packages\neo\rawio\spikegadgetsrawio.py", line 227, in _parse_header
    signal_channels = signal_channels[keep]
IndexError: list index out of range
```

**Environment:**
 - OS: Windows
 - Python 3.9
 - Neo 0.13.1dev1
 - NumPy 1.26.4

### PR DESCRIPTION
**Pull request by zm711**
Fixes #1517 

Howdy @RobertoDF,

I think this should fix your problem :) 

As I expected with neuropixels you can have different channel/chip ratios then we expected with the intan based chip. I think your colleague's data just happened to line up with the intan expectations. So in this case, I check if the headstage fits our expectations and if it doesn't I revert back to the old way of parsing the data. Note that I don't see your gain problem. Looking at a few random channels I see this:

```
  ('trode1252chan341', '341', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1251chan340', '340', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1250chan277', '277', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1249chan276', '276', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1248chan213', '213', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1247chan212', '212', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1246chan149', '149', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1245chan148', '148', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1244chan85', '85', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1243chan84', '84', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1242chan21', '21', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1241chan20', '20', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1240chan723', '723', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1239chan722', '722', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1238chan659', '659', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1237chan658', '658', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
        ('trode1236chan595', '595', 30000., 'int16', 'uV', 0.01831111, 0., 'trodes'),
```

The gain is correct.

### CODE REVIEW

**Comment by zm711** [comment]
This parsed the file you shared with me on my computer and seemed correct based on my summary look at the data. I'll likely push a couple more commits to add some comments to the code etc, but I should only be making cosmetic changes at this point.

**Comment by RobertoDF** [comment]
Just tested it and it works. Amazing! Thanks for fixing it so quickly.

**Comment by zm711** [comment]
That's awesome. I'll ping @alejoe91 for a review. But let's wait until end of next week before we merge at minimum so we can hopefully get a test file from you :)

File: neo/rawio/spikegadgetsrawio.py
Comment (zm711) [inline neo/rawio/spikegadgetsrawio.py:128]:
```suggestion
        self._trode_version = gconf.attrib["trodesVersion"]
```
oops

**Comment by zm711** [comment]
@samuelgarcia, now that you're back you could also look over this. Basically, it spikegadgets doesn't always use an intan chip. i couldn't see any way in the metadata to tell which headstage is used, so if you have a better idea feel free to push, but this prevents the reader from breaking when using neuropixel style data instead of intan headstage style data.

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**
